[examples] Adds hot potato example (#3658)
MystenLabs/sui: doc/book/src/patterns/hot-potato.md

# Hot Potato

> Hot Potato is a name for a struct that has no abilities, hence it can only packed and unpacked in its module. The pattern itself creates a case where function B has to be called after function A if A returns a potato and B consumes it.

Reviewer: ```suggestion
Hot Potato is a name for a struct that has no abilities, hence it can only be packed and unpacked in its module. In this struct, you must call function B after function A in the case where function A returns a potato and function B consumes it.
```
The second sentence is a bit hard to parse, hopefully my proposed change is still accurate.
